### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by Noorae**
### Various UI related fixes and refactoring

- Fixed bug #127 
     SwipeableDrawer is now only rendered if isMobile and isCoarsePointer is true, otherwise Desktop drawer is used. This means that the Mobile drawer is used only on touch screen devices.

- SwipeableDrawer height fixed to fit all content of the drawer, even on smaller screens.

- Added background coloring for date values in the DateSlider for better view:

     image

- Various state manipulations to make the slider react to DatePicker changes,  comparisonview and SatelittleView and to make these components update with each other.

    -    User can now use the buttons or the datePickers to manipulate which view they want to keep on and what dates to display. 
    -    If any bugs encountered, raise a new bug issue as the states could be refactored even further to make the logic more simple. 
    -    Added comments on some of the updating logic to help refactor later if necessary.
    -    Some empty useEffects added to make sure components rerender when necessary.

- added localisation to client components

### CODE REVIEW

File: app/components/controls/layerControl/LayerControl.tsx [hunk 1]
```diff
@@ -44,11 +46,14 @@ export default function LayerControl({
       overlayVisibility.observations && zoomLevel < FETCH_ZOOM_LEVEL_THRESHOLD;
 
     if (sightingsError && observationsError) {
-      setFetchingError("Zoomaa lähemmäksi tietojen hakemista varten");
+      const error = t("sightingsObservationsError");
```
Comment (kuukkanen) [inline app/components/controls/layerControl/LayerControl.tsx:49]: You're using `t` here, so should it be part of the `useEffect` dependencies?

File: app/components/map/comparison/SideBySide.tsx [hunk 2]
```diff
@@ -18,6 +18,8 @@ export default function SideBySide({
 }: SideBySideProps) {
   const map = useMap();
 
+  useEffect(() => {}, [leftDate, rightDate]);
```
Comment (kuukkanen) [inline app/components/map/comparison/SideBySide.tsx:21]: Is this necessary? You have `leftDate` and `rightDate` as dependencies in the `useEffect` below.

File: app/utils/i18n.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
 export default {
   // This is the list of languages your application supports
-  supportedLngs: ["fi"],
+  supportedLngs: ["fi", "en"],
```
Comment (kuukkanen) [inline app/utils/i18n.ts:3]: We don't have English translations, now do we?

Comment (kuukkanen) [reply app/utils/i18n.ts:3]: I'm not sure if it's needed by i18next, but we should not specify any languages that we don't have translations for.

**Review by kuukkanen — CHANGES_REQUESTED** [review]

File: app/utils/i18n.ts (hunk 3 above)
Comment (Noorae) [reply app/utils/i18n.ts:3]: good point, it needs to be removed. There was an error with the finnish translations not working, adding it when the fallback language was still english, made an error disappear, but it should have been removed when fallback languages was switched to finnish.

File: app/components/controls/layerControl/LayerControl.tsx (hunk 1 above)
Comment (Noorae) [reply app/components/controls/layerControl/LayerControl.tsx:49]: added to dependencies, good catch.

File: app/components/map/comparison/SideBySide.tsx (hunk 2 above)
Comment (Noorae) [reply app/components/map/comparison/SideBySide.tsx]: removed

File: app/utils/i18n.ts (hunk 3 above)
Comment (Noorae) [reply app/utils/i18n.ts]: english removed

**Review by kuukkanen — APPROVED** [review]
Nice work! 👍🏻

**Pull request merged**